src/traces/mesh3d/defaults.js

```diff
@@ -34,19 +34,21 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     }
 
     var coords = readComponents(['x', 'y', 'z']);
-    var indices = readComponents(['i', 'j', 'k']);
-
     if(!coords) {
         traceOut.visible = false;
         return;
     }
 
-    if(indices) {
-        // otherwise, convert all face indices to ints
-        indices.forEach(function(index) {
-            for(var i = 0; i < index.length; ++i) index[i] |= 0;
-        });
+    // three indices should be all provided or not
+    if(
+        (traceIn.i && (!traceIn.j || !traceIn.k)) ||
```

Comment:
Ha, you're using `traceIn` here. So something like

```
{
  i: 1,
  j: true,
  k: {}
}
```

would have `visible:true`.

Could we be more stringent and use `traceOut` instead, to ensure that i,j,k are all non-empty arrays.

Reply:
Good Call. Now `traceOut` is applied.
Noting that `visible` would "only" be `false` in case of bad 'vertices'.

Reply:
Hmm. I don't understand `traceOut.i`, `traceOut.j` and `traceOut.k` should be defined at this stage. They get filled in during the `coerce()` calls in `readComponents` below.

Reply:
You are right. That is fixed in https://github.com/plotly/plotly.js/pull/3369/commits/29f5a06c2aa5be4101c33ed649f8ef74a19a0d71
Now the traces become invisible in case of bad index arrays.